sympy/codegen/ast.py

```diff
@@ -481,15 +524,73 @@ def __hash__(self):
         return super(Token, self).__hash__()
 
     def _sympystr(self, printer):
-        return "{0}({1})".format(self.__class__.__name__, ', '.join(
-            ['%s=%s' % (k, printer._print(getattr(self, k))) for k in self.__slots__]
-        ))
+        values = [getattr(self, k) for k in self.__slots__]
+        return "{0}({1})".format(self.__class__.__name__, ', '.join([
+            ('%s=' + ('"%s"' if isinstance(v, string_types) else '%s')) % (k, printer._print(v))
+            for k, v in zip(self.__slots__, values)
+        ]))
+
+    def kwargs(self, exclude=(), apply=lambda arg: arg):
+        return {k: apply(getattr(self, k)) for k in self.__slots__ if k not in exclude}
+
+
+class NoneToken(Token):
+
+    def __eq__(self, other):
+        return other is None or isinstance(other, NoneToken)
+
+    def _hashable_content(self):
+        return ()
+
+    def __hash__(self):
+        return super(Token, self).__hash__()
+
+
+none = NoneToken()
```

Comment:
Using this "trick" (together with `class String`) allowed me to use `Basic.args` throughout―now SymPy's `replace` works as expected. @asmeurer I hope this is acceptable (it only lives in .codegen.ast after all)?

Reply:
Turns out this is also useful for representing the value `None` when printing Python code using `pycode`.